domain/resource/doc.go

```diff
@@ -1,8 +1,27 @@
 // Copyright 2024 Canonical Ltd.
 // Licensed under the AGPLv3, see LICENCE file for details.
 
-// Package resource provides the domain types for a resource. The application
-// domain is the primary domain for creation and handling of a resource. The
-// types here are to ensure that the resource is handled correctly and that
-// the resource is correctly represented in the domain.
+// Package resource provides the domain types for handling charm resources once
+// associated with application.
+//
+// Resources represent supplementary files or configurations associated with a charm
+// during deployment. They are defined through the charm, then exist in three
+// forms after deployment:
+//
+//   - application resources are resources bounded with an application and
+//     defined at a specific revision (namely "available" resources)
+//
+//   - unit resources are resources downloaded and used by a unit
+//
+//   - repository resources represents last revision available in charm
+//     repository (namely "potential")
+//
+// Resources can be refreshed independently of charm revisions.
```

Comment:
// Resources can be refreshed independently of charm revisions, but must exist in the current charm version used by the application.